source/compiler/sc3.c

```diff
@@ -1832,8 +1832,6 @@ static int hier1(value *lval1)
       if (lval2.ident==iCONSTEXPR && lval2.sym!=NULL
           && lval2.sym->dim.array.length>0 && sym->dim.array.level==0)
       {
-        lval1->tag=lval2.sym->x.tags.index;
-        lval1->constval=lval2.sym->dim.array.length;
         if (lval2.tag==sym->x.tags.index && lval1->constval>1 && matchtoken('[')) {
```

Comment:
I removed these two lines, as they turned out to be duplicate code. There's already code that does the same thing (saves the tag and the enum field size into the resulting value) at lines 1784-1785.